cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -795,10 +802,17 @@ private Put getIndexPut(MetadataEntity metadataEntity, String metadataKey, Strin
    * @param metadataEntity target id for which metadata needs snapshotting
    */
   private void writeHistory(MetadataEntity metadataEntity) {
+    writeHistory(metadataEntity, -1);
+  }
+
+  private void writeHistory(MetadataEntity metadataEntity, long time) {
```

Comment:
it looks like there is no place in your code, where you actually call this, except for right above. So time is always -1? Then why is this parameter needed?